docs/migration_to_1.0.0.md

```diff
@@ -0,0 +1,72 @@
+# Migrating from Wasmer 0.x to Wasmer 1.0.0
+
+Wasmer 1.0.0 is currently in alpha and is our primary focus. This document will
+describe the differences between 0.x Wasmer and Wasmer 1.0.0 and provide examples
+to make migrating to the new API as simple as possible.
+
+Some features are still under development during the alpha of Wasmer 1.0.0. This document
+will aim to make clear what these features are.
+
+## Table of Contents
+
+- Rationale for changes in 1.0.0
+- How to use Wasmer 1.0.0
+- Project structure
+- TODO: specific differences
+
+## Rationale for changes in 1.0.0
+
+TODO: explain why 1.0.0 exists
+
+## How to use Wasmer 1.0.0
+
+### Installing Wasmer
+
+See [wasmer.io] for installation instructions.
+
+If you already have wasmer installed, run `wasmer self-update`.
+
+Install the latest versions of wasmer with [wasmer-nightly].
+
+### Using Wasmer 1.0.0
+
+The CLI interface for Wasmer 1.0.0 is the same as it was in Wasmer 0.X.
```

Comment:
The interface has varied a bit (although none are breaking changes, since the old syntax is also supported).

Before users did:
```bash
wasmer run myfile.wasm --backend=cranelift
```

And now they do:
```bash
wasmer run myfile.wasm --cranelift
```

Reply:
good catch!